Clockwise.Tests/CircuitBreakerTests.cs

```diff
@@ -0,0 +1,124 @@
+using Xunit;
+
+namespace Clockwise.Tests
+{
+    public class ACICircuitBreaker : CircuitBreaker<ACICircuitBreaker>
+    {
+        public ACICircuitBreaker(ICircuitBreakerStorage storage) : base(storage)
+        {
+        }
+    }
+    public class CircuitBreakerTests
+    {
+        [Fact]
+        public void When_circuitbreaker_is_not_configured_and_the_handler_requests_the_pause_then_usefull_exception_is_thrown()
+        {
+
+            var cfg = new Configuration()
+                .UseInMemoryScheduling()
+                .UseCircuitbreakerFor<int, ACICircuitBreaker>();
+            
+            new Action(() =>
+            {
+                cfg.CommandReceiver<int>();
+            }).Should().Throw<ConfigurationException>();
+
+        }
+
+        [Fact]
+        public async Task When_failure_occours_then_the_handler_can_signal_the_circuitbreaker()
+        {
+            using (var clock = VirtualClock.Start())
+            {
+                var processed = new List<int>();
+                var cfg = new Configuration();
+                cfg = cfg
+                    .UseDependency<ICircuitBreakerStorage>(type => new InMemoryCircuitBreakerStorage())
+                    .UseInMemoryScheduling()
+                   .UseCircuitbreakerFor<int, ACICircuitBreaker>();
+
+                var handler = CommandHandler.Create<int>(delivery =>
+                {
+                    if (delivery.Command > 10)
+                    {
+                        return delivery.PauseAllDeliveriesFor(5.Seconds());
+                    }
+
+                    processed.Add(delivery.Command);
+                    return delivery.Complete();
+                });
+
+                var rc = cfg.CommandReceiver<int>();
+                rc.Subscribe(handler);
+
+                var scheduler = cfg.CommandScheduler<int>();
+                await scheduler.Schedule(1, 1.Seconds());
+                await scheduler.Schedule(2, 2.Seconds());
+                await scheduler.Schedule(11, 3.Seconds());
+                await scheduler.Schedule(3, 4.Seconds());
+
+                await clock.AdvanceBy(5.Seconds());
+
+                processed.Should().BeEquivalentTo(1, 2);
+
+                await clock.AdvanceBy(6.Seconds());
+
+                processed.Should().BeEquivalentTo(1, 2, 3);
+            }
+        }
+
+        [Fact]
+        public async Task When_failure_occours_then_no_handler_for_same_command_receive_messages()
+        {
+            using (var clock = VirtualClock.Start())
+            {
+                var processedInt = new List<int>();
+                var processedLong = new List<long>();
+                var cfg = new Configuration();
+                cfg = cfg
+                    .UseInMemeoryCircuitBreakerStorage()
+                    .UseInMemoryScheduling()
+                    .UseCircuitbreakerFor<int, ACICircuitBreaker>()
+                    .UseCircuitbreakerFor<long, ACICircuitBreaker>();
```

Comment:
I think this can be named better. The word "for" is a little confusing because there are now two generic type parameters.